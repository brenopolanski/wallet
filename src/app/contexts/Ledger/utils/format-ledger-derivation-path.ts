import { LedgerDerivationScheme } from "app/contexts/Ledger/contracts";

export const formatLedgerDerivationPath = (scheme: LedgerDerivationScheme) =>
	`m/${scheme.purpose || 44}'/${scheme.coinType}'/${scheme.account || 0}'/${scheme.change || 0}/${
		scheme.address || 0
	}`;
