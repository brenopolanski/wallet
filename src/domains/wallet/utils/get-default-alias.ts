import { Networks } from "@payvo/sdk";
import { Contracts } from "@payvo/sdk-profiles";

interface GetDefaultAliasInput {
	profile: Contracts.IProfile;
	network: Networks.Network;
}

export const getDefaultAlias = ({ profile, network }: GetDefaultAliasInput): string => {
	const makeAlias = (count: number) => `${network.displayName()} #${count}`;

	const sameCoinWallets = profile.wallets().findByCoinWithNetwork(network.coin(), network.id());

	let sameCoinWalletsCount = sameCoinWallets.length || 1;

	while (profile.wallets().findByAlias(makeAlias(sameCoinWalletsCount))) {
		sameCoinWalletsCount++;
	}

	return makeAlias(sameCoinWalletsCount);
};
