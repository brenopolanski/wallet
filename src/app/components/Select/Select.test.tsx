import React from "react";
import { fireEvent, render, screen } from "utils/testing-library";

import { Select } from "./Select";

describe("Select", () => {
	it("should render a select input with placeholder", () => {
		const placeholder = "Select option";
		const { asFragment } = render(<Select placeholder="Select option" />);

		expect(screen.getByText(placeholder)).toBeInTheDocument();
		expect(asFragment()).toMatchSnapshot();
	});

	it("should render a select input with options", () => {
		const { asFragment } = render(
			<Select defaultValue={1}>
				<option data-testid="select-option" value={1}>
					Option 1
				</option>
				<option data-testid="select-option" value={2}>
					Option 2
				</option>
			</Select>,
		);

		fireEvent.change(screen.getByTestId("Select"), { target: { value: 2 } });
		const options = screen.getAllByTestId("select-option");

		expect((options[0] as HTMLOptionElement).selected).toBeFalsy();
		expect((options[1] as HTMLOptionElement).selected).toBeTruthy();

		expect(asFragment()).toMatchSnapshot();
	});
});
