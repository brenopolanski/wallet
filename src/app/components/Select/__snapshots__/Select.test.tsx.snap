// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select should render a select input with options 1`] = `
<DocumentFragment>
  <div
    class="Input__InputWrapperStyled-sc-1i8sd1r-0 gXyrpF"
  >
    <div
      class="relative flex flex-1 h-full"
    >
      <select
        class="Input__InputStyled-sc-1i8sd1r-1 hPntwa p-0 border-none bg-transparent focus:ring-0 no-ligatures w-full"
        data-testid="Select"
      >
        <option
          data-testid="select-option"
          selected=""
          value="1"
        >
          Option 1
        </option>
        <option
          data-testid="select-option"
          value="2"
        >
          Option 2
        </option>
      </select>
    </div>
    <div
      class="flex items-center space-x-3 divide-x divide-theme-secondary-300 dark:divide-theme-secondary-800 text-theme-primary-300 dark:text-theme-secondary-600"
    >
      <div
        class=""
      >
        <span
          class="w-12 text-lg pointer-events-none text-theme-secondary-text"
        >
          <div
            class="sc-bdvvtL Cfvxc"
            height="10"
            width="10"
          >
            <svg>
              chevron-down-small.svg
            </svg>
          </div>
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Select should render a select input with placeholder 1`] = `
<DocumentFragment>
  <div
    class="Input__InputWrapperStyled-sc-1i8sd1r-0 gXyrpF"
  >
    <div
      class="relative flex flex-1 h-full"
    >
      <select
        class="Input__InputStyled-sc-1i8sd1r-1 hPntwa p-0 border-none bg-transparent focus:ring-0 no-ligatures w-full"
        data-testid="Select"
      >
        <option
          disabled=""
          selected=""
          value=""
        >
          Select option
        </option>
      </select>
    </div>
    <div
      class="flex items-center space-x-3 divide-x divide-theme-secondary-300 dark:divide-theme-secondary-800 text-theme-primary-300 dark:text-theme-secondary-600"
    >
      <div
        class=""
      >
        <span
          class="w-12 text-lg pointer-events-none text-theme-secondary-text"
        >
          <div
            class="sc-bdvvtL Cfvxc"
            height="10"
            width="10"
          >
            <svg>
              chevron-down-small.svg
            </svg>
          </div>
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
